src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/utils/rgw-bucket-tiering.ts

```diff
@@ -32,7 +32,30 @@ export class BucketTieringUtils {
 
   private static getTierTargets(tierTarget: TierTarget, zoneGroup: string, targetName: string) {
     const val = tierTarget.val;
-    if (val.tier_type === TIER_TYPE.CLOUD_TIER) {
+
+    if (tierTarget.val.tier_type === TIER_TYPE.GLACIER) {
+      return {
+        zonegroup_name: zoneGroup,
+        placement_target: targetName,
+        storage_class: tierTarget.val.storage_class,
+        retain_head_object: tierTarget.val.retain_head_object,
+        allow_read_through: tierTarget.val.allow_read_through,
+        glacier_restore_days: tierTarget.val['s3-glacier'].glacier_restore_days,
+        glacier_restore_tier_type: tierTarget.val['s3-glacier'].glacier_restore_tier_type,
+        restore_storage_class: tierTarget.val.restore_storage_class,
+        read_through_restore_days: tierTarget.val.read_through_restore_days,
+        tier_type: tierTarget.val.tier_type,
+        region: tierTarget.val.s3.region,
+        endpoint: tierTarget.val.s3.endpoint,
+        target_path: tierTarget.val.s3.target_path,
+        access_key: tierTarget.val.s3.access_key,
+        secret: tierTarget.val.s3.secret,
+        multipart_min_part_size: tierTarget.val.s3.multipart_min_part_size,
+        multipart_sync_threshold: tierTarget.val.s3.multipart_sync_threshold,
+        host_style: tierTarget.val.s3.host_style,
+        ...tierTarget.val['s3-glacier']
```

Comment:
can't we use something like

```suggestion
       zonegroup_name: zoneGroup,
        placement_target: targetName,
        storage_class: val.storage_class,
        retain_head_object: val.retain_head_object,
        allow_read_through: val.allow_read_through,
        tier_type: val.tier_type,
        ...val.s3,
        ...val['s3-glacier']
```